VulkanHppGenerator.cpp

```diff
@@ -742,7 +742,7 @@ const std::string deleterClassString = R"(
       ObjectDestroy( OwnerType owner = OwnerType(), Optional<const AllocationCallbacks> allocator = nullptr, Dispatch const &dispatch = Dispatch() )
         : m_owner( owner )
         , m_allocator( allocator )
-        , m_dispatch( dispatch )
+        , m_dispatch( &dispatch )

```

Comment:
This seems dangerous: aren’t you going to end up with a dangling pointer if the constructor is called without providing a `Dispatch` object, as you would be taking the address of a temporary? Or if the `dispatch` object goes out-of-scope before `ObjectDestroy` does? (For example `dispatch` is a local variable in a class method, and `ObjectDestroy`, indirectly via `UniqueInstance` for example, is an attribute of that same class.)

The existing code has the same issue.

Reply:
Yes, I noticed that too, but couldn't find an obvious/easy fix, since vk::DispatchLoaderDynamic would be too big to reasonably copy.
The best option I can currently think of would be using a static instance as a default parameter value. e.g.:

```cpp
namespace detail {
	template<class Dispatch>
	const Dispatch& defaultDispatchInstance() {
		static const Dispatch instance;
		return instance;
	}
}

// ...

ObjectFree( OwnerType owner = OwnerType(), Optional<const AllocationCallbacks> allocator = nullptr, Dispatch const &dispatch = detail::defaultDispatchInstance<Dispatch>() )
```

Reply:
There might still be some issues for DispatchLoaderDynamic, as you might want to give it a special instance and device.

Another approach, though it’s a bit heavy, would be to change the API to take a shared_ptr to the dispatch object, instead of the reference; that way you are sure it won’t run away while you’re still holding on it.